### PR TITLE
Integrate jpsreport back into jpscore

### PR DESCRIPTION
**Pull request by Ozaq**
Jpsreport relies on several structures from jpscore, code was copied from
core to report and already started to drift apart. Since those projects
are tightly coppled it makes life a lot easier to track them in one
repository.

This is currently a draft that I discussed with @zeroset. One thing remaining before this could potentially be merged is loading of geometry files. The code around this has changed so between both versions of `Building.cpp` and needs to be fixed.

### CODE REVIEW

**Comment by gjaeger** [comment]
We should transfer the current state of the jpsreport. This draft contains an obsolete state. We should clarify when to stop working in the repository of jpsreport.

**Comment by Ozaq** [comment]
> We should transfer the current state of the jpsreport. This draft contains an obsolete state. We should clarify when to stop working in the repository of jpsreport.

Hi @gjaeger yeah I noticed today, took me some time to arrive here and I didn't want to make it more complicated by updating. I saw I am missing 6 commits as of today. @zeroset wanted to get in touch with you asap to figure out if jpsreport works as desired on this PR. If thats the case I can transfer the missing changes. For reference: jpsreport on this branch is taken from commit 66ca522e13bc4f92d75da08165d146ad1fbac43d

File: jpsreport/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,80 @@
+################################################################################
+# Build jspreport executable
+################################################################################
+add_executable(jpsreport
+    src/Analysis.cpp
+    src/Analysis.h
+    src/ArgumentParser.cpp
+    src/ArgumentParser.h
+    src/buildingOld.cpp
```
Comment (zeroset) [inline jpsreport/CMakeLists.txt]:
```suggestion
    src/BuildingOld.cpp
```

File: jpsreport/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    src/buildingOld.h
```
Comment (zeroset) [inline jpsreport/CMakeLists.txt]:
```suggestion
    src/BuildingOld.h
```

Comment (Ozaq) [reply jpsreport/CMakeLists.txt]: fixed

File: jpsreport/CMakeLists.txt (hunk 1 above)
Comment (Ozaq) [reply jpsreport/CMakeLists.txt]: fixed

**Comment by Ozaq** [comment]
Integrated latest changes from jpsreport (7714d4a5ecadd3cca11544c1d5a6509ccbacf459) 
NOTE: Testdata / tests are not yet migrated

**Comment by gjaeger** [comment]
with commit bdef837:

*CMake*
```bash
/Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH=/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/deps -G "CodeBlocks - Unix Makefiles" /Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- PROJECT_VERSION=0.8.4
-- BUILD_TESTING=OFF
-- BUILD_CPPUNIT_TEST=OFF
-- BUILD_DOC=OFF
-- JPSFIRE=OFF
-- AIROUTER=OFF
-- USE_OPENMP=ON
-- BUILD_WITH_ASAN=OFF
CMake Warning at /usr/local/lib/cmake/boost_filesystem-1.71.0/libboost_filesystem-variant-shared.cmake:59 (message):
  Target Boost::filesystem already has an imported location
  '/usr/local/lib/libboost_filesystem-mt.dylib', which will be overwritten
  with '/usr/local/lib/libboost_filesystem.dylib'
Call Stack (most recent call first):
  /usr/local/lib/cmake/boost_filesystem-1.71.0/boost_filesystem-config.cmake:43 (include)
  /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:117 (find_package)
  /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:182 (boost_find_component)
  /Applications/CLion.app/Contents/bin/cmake/mac/share/cmake-3.15/Modules/FindBoost.cmake:443 (find_package)
  CMakeLists.txt:108 (find_package)


-- Found Boost: /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake (found suitable version "1.71.0", minimum required is "1.65") found components:  filesystem 
-- Found OpenMP: /usr/local/lib/libomp.dylib  
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found PythonInterp: /usr/local/bin/python3 (found suitable version "3.7.4", minimum required is "3") 
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/cmake-build-debug

[Finished]
```

*Make*
```bash
====================[ Build | all | Debug ]=====================================
/Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake --build /Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/cmake-build-debug --target all -- -j 2
Scanning dependencies of target visilibity
Scanning dependencies of target poly2tri
[  1%] Building CXX object third-party/CMakeFiles/visilibity.dir/visiLibity/source_code/visilibity.cpp.o
[  1%] Building CXX object third-party/CMakeFiles/poly2tri.dir/poly2tri/common/shapes.cpp.o
[  2%] Building CXX object third-party/CMakeFiles/poly2tri.dir/poly2tri/sweep/advancing_front.cpp.o
[  3%] Building CXX object third-party/CMakeFiles/poly2tri.dir/poly2tri/sweep/cdt.cpp.o
[  4%] Building CXX object third-party/CMakeFiles/poly2tri.dir/poly2tri/sweep/sweep.cpp.o
[  5%] Building CXX object third-party/CMakeFiles/poly2tri.dir/poly2tri/sweep/sweep_context.cpp.o
[  6%] Linking CXX static library ../lib/libvisilibity.a
[  6%] Built target visilibity
Scanning dependencies of target tinyxml
[  7%] Building CXX object third-party/CMakeFiles/tinyxml.dir/tinyxml/tinystr.cpp.o
[  8%] Building CXX object third-party/CMakeFiles/tinyxml.dir/tinyxml/tinyxml.cpp.o
[  9%] Linking CXX static library ../lib/libpoly2tri.a
[  9%] Built target poly2tri
[ 10%] Building CXX object third-party/CMakeFiles/tinyxml.dir/tinyxml/tinyxmlerror.cpp.o
[ 11%] Building CXX object third-party/CMakeFiles/tinyxml.dir/tinyxml/tinyxmlparser.cpp.o
[ 12%] Linking CXX static library ../lib/libtinyxml.a
[ 12%] Built target tinyxml
Scanning dependencies of target core
[ 12%] Building CXX object libcore/CMakeFiles/core.dir/events/Event.cpp.o
[ 13%] Building CXX object libcore/CMakeFiles/core.dir/events/EventManager.cpp.o
[ 14%] Building CXX object libcore/CMakeFiles/core.dir/general/ArgumentParser.cpp.o
[ 15%] Building CXX object libcore/CMakeFiles/core.dir/general/Filesystem.cpp.o
[ 16%] Building CXX object libcore/CMakeFiles/core.dir/general/Logger.cpp.o
In file included from /Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/libcore/general/ArgumentParser.cpp:21:
/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/libcore/general/ArgumentParser.h:48:19: warning: private field 'iniFilePathOpt' is not used [-Wunused-private-field]
    CLI::Option * iniFilePathOpt =
                  ^
/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/libcore/general/ArgumentParser.h:51:19: warning: private field 'logLevelOpt' is not used [-Wunused-private-field]
    CLI::Option * logLevelOpt =
                  ^
[ 17%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Building.cpp.o
2 warnings generated.
[ 18%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Crossing.cpp.o
[ 19%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Goal.cpp.o
[ 20%] Building CXX object libcore/CMakeFiles/core.dir/geometry/GoalManager.cpp.o
[ 21%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Hline.cpp.o
[ 22%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Line.cpp.o
[ 23%] Building CXX object libcore/CMakeFiles/core.dir/geometry/NavLine.cpp.o
[ 23%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Obstacle.cpp.o
[ 24%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Point.cpp.o
[ 25%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Room.cpp.o
[ 26%] Building CXX object libcore/CMakeFiles/core.dir/geometry/SubRoom.cpp.o
[ 27%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Transition.cpp.o
[ 28%] Building CXX object libcore/CMakeFiles/core.dir/geometry/WaitingArea.cpp.o
[ 29%] Building CXX object libcore/CMakeFiles/core.dir/geometry/Wall.cpp.o
[ 30%] Building CXX object libcore/CMakeFiles/core.dir/IO/GeoFileParser.cpp.o
[ 31%] Building CXX object libcore/CMakeFiles/core.dir/IO/IniFileParser.cpp.o
[ 32%] Building CXX object libcore/CMakeFiles/core.dir/IO/OutputHandler.cpp.o
[ 33%] Building CXX object libcore/CMakeFiles/core.dir/IO/PedDistributionParser.cpp.o
[ 34%] Building CXX object libcore/CMakeFiles/core.dir/IO/progress_bar.cpp.o
[ 34%] Building CXX object libcore/CMakeFiles/core.dir/IO/Trajectories.cpp.o
[ 35%] Building CXX object libcore/CMakeFiles/core.dir/IO/TraVisToClient.cpp.o
[ 36%] Building CXX object libcore/CMakeFiles/core.dir/math/ForceModel.cpp.o
[ 37%] Building CXX object libcore/CMakeFiles/core.dir/math/GCFMModel.cpp.o
[ 38%] Building CXX object libcore/CMakeFiles/core.dir/math/GompertzModel.cpp.o
[ 39%] Building CXX object libcore/CMakeFiles/core.dir/math/GradientModel.cpp.o
[ 40%] Building CXX object libcore/CMakeFiles/core.dir/math/KrauszModel.cpp.o
[ 41%] Building CXX object libcore/CMakeFiles/core.dir/math/Mathematics.cpp.o
[ 42%] Building CXX object libcore/CMakeFiles/core.dir/math/ODESolver.cpp.o
[ 43%] Building CXX object libcore/CMakeFiles/core.dir/math/OperationalModel.cpp.o
[ 44%] Building CXX object libcore/CMakeFiles/core.dir/math/VelocityModel.cpp.o
[ 45%] Building CXX object libcore/CMakeFiles/core.dir/mpi/LCGrid.cpp.o
[ 45%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/AgentsParameters.cpp.o
[ 46%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/AgentsQueue.cpp.o
[ 47%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/AgentsSource.cpp.o
[ 48%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/AgentsSourcesManager.cpp.o
[ 49%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/Ellipse.cpp.o
[ 50%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/Knowledge.cpp.o
[ 51%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/PedDistributor.cpp.o
[ 52%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/Pedestrian.cpp.o
[ 53%] Building CXX object libcore/CMakeFiles/core.dir/pedestrian/StartDistribution.cpp.o
[ 54%] Building CXX object libcore/CMakeFiles/core.dir/routing/DirectionStrategy.cpp.o
[ 55%] Building CXX object libcore/CMakeFiles/core.dir/routing/ff_router/ffRouter.cpp.o
[ 56%] Building CXX object libcore/CMakeFiles/core.dir/routing/ff_router/FloorfieldViaFM.cpp.o
[ 56%] Building CXX object libcore/CMakeFiles/core.dir/routing/ff_router/UnivFFviaFM.cpp.o
[ 57%] Building CXX object libcore/CMakeFiles/core.dir/routing/global_shortest/AccessPoint.cpp.o
[ 58%] Building CXX object libcore/CMakeFiles/core.dir/routing/global_shortest/DTriangulation.cpp.o
[ 59%] Building CXX object libcore/CMakeFiles/core.dir/routing/global_shortest/GlobalRouter.cpp.o
[ 60%] Building CXX object libcore/CMakeFiles/core.dir/routing/quickest/QuickestPathRouter.cpp.o
[ 61%] Building CXX object libcore/CMakeFiles/core.dir/routing/Router.cpp.o
[ 62%] Building CXX object libcore/CMakeFiles/core.dir/routing/RoutingEngine.cpp.o
[ 63%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/Brain.cpp.o
[ 64%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/BrainStorage.cpp.o
[ 65%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/associations.cpp.o
[ 66%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/cogmapoutputhandler.cpp.o
[ 67%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/cognitivemap.cpp.o
[ 67%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/connection.cpp.o
[ 68%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/internnavigationnetwork.cpp.o
[ 69%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/landmark.cpp.o
[ 70%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/landmarknetwork.cpp.o
[ 71%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/region.cpp.o
[ 72%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/cognitiveMap/youareherepointer.cpp.o
[ 73%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/GraphNetwork.cpp.o
[ 74%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/navigation_graph/GraphEdge.cpp.o
[ 75%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/navigation_graph/GraphVertex.cpp.o
[ 76%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/NavigationGraph.cpp.o
[ 77%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/sensor/AbstractSensor.cpp.o
[ 78%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/sensor/DiscoverDoorsSensor.cpp.o
[ 78%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/sensor/JamSensor.cpp.o
[ 79%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/sensor/LastDestinationsSensor.cpp.o
[ 80%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/sensor/locater.cpp.o
[ 81%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/sensor/RoomToFloorSensor.cpp.o
[ 82%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/sensor/SensorManager.cpp.o
[ 83%] Building CXX object libcore/CMakeFiles/core.dir/routing/smoke_router/SmokeRouter.cpp.o
[ 84%] Building CXX object libcore/CMakeFiles/core.dir/Simulation.cpp.o
[ 85%] Building CXX object libcore/CMakeFiles/core.dir/voronoi-boost/VoronoiPositionGenerator.cpp.o
[ 86%] Linking CXX static library ../lib/libcore.a
[ 86%] Built target core
Scanning dependencies of target jpscore
Scanning dependencies of target jpsreport
[ 87%] Building CXX object jpscore/CMakeFiles/jpscore.dir/src/main.cpp.o
[ 88%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/Analysis.cpp.o
[ 88%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/ArgumentParser.cpp.o
[ 89%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/BuildingOld.cpp.o
[ 90%] Linking CXX executable ../bin/jpscore
[ 90%] Built target jpscore
[ 91%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/geometry_helper.cpp.o
[ 92%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/main.cpp.o
[ 93%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/methods/Method_A.cpp.o
[ 94%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/methods/Method_B.cpp.o
[ 95%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/methods/Method_C.cpp.o
[ 96%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/methods/Method_D.cpp.o
[ 97%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/methods/Method_I.cpp.o
/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/jpsreport/src/methods/Method_I.cpp:606:51: warning: unused parameter 'individualFD' [-Wunused-parameter]
     void Method_I::SetCalculateIndividualFD(bool individualFD)
                                                  ^
1 warning generated.
[ 98%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/methods/PedData.cpp.o
[ 99%] Building CXX object jpsreport/CMakeFiles/jpsreport.dir/src/methods/VoronoiDiagram.cpp.o
[100%] Linking CXX executable ../bin/jpsreport
[100%] Built target jpsreport

Build finished
```

**Comment by gjaeger** [comment]
There is a problem with `jpreport`:

```bash
jpscore/cmake-build-debug/bin/jpsreport

INFO: 	Trying to load the default configuration from the file <ini.xml>
INFO: 	Parsing the ini file <ini.xml>
ERROR: 	Failed to open file
ERROR: 	Could not parse the ini file
Usage:

/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/cmake-build-debug/bin/jpsreport inifile.xml
```

I miss the header.

**Comment by gjaeger** [comment]
During this test for Methods A to I ([integrate_jpsreport_into_jpscore.zip](https://github.com/JuPedSim/jpscore/files/3793293/integrate_jpsreport_into_jpscore.zip)) I get the following message:

```
% /Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/cmake-build-debug/bin/jpsreport ini_Methods.xml

INFO: 	Parsing the ini file <ini_Methods.xml>
INFO:	logfile </Users/gjaeger/Desktop/integrate_jpsreport_into_jpscore/logfile_Methods.txt>INFO:	Success with Method A using measurement area id 2

INFO:	Success with Method A using measurement area id 6

INFO:	Success with Method B using measurement area id 4
INFO:	Success with Method C using measurement area id 4
INFO:	Success with Method C using measurement area id 3
libc++abi.dylib: terminating with uncaught exception of type boost::numeric::positive_overflow: bad numeric conversion: positive overflow
zsh: abort       ini_Methods.xml
```

**Comment by Ozaq** [comment]
> During this test for Methods A to I ([integrate_jpsreport_into_jpscore.zip](https://github.com/JuPedSim/jpscore/files/3793293/integrate_jpsreport_into_jpscore.zip)) I get the following message:
> 
> ```
> % /Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/cmake-build-debug/bin/jpsreport ini_Methods.xml
> 
> INFO: 	Parsing the ini file <ini_Methods.xml>
> INFO:	logfile </Users/gjaeger/Desktop/integrate_jpsreport_into_jpscore/logfile_Methods.txt>INFO:	Success with Method A using measurement area id 2
> 
> INFO:	Success with Method A using measurement area id 6
> 
> INFO:	Success with Method B using measurement area id 4
> INFO:	Success with Method C using measurement area id 4
> INFO:	Success with Method C using measurement area id 3
> libc++abi.dylib: terminating with uncaught exception of type boost::numeric::positive_overflow: bad numeric conversion: positive overflow
> zsh: abort       ini_Methods.xml
> ```

Thanks for testing! I will have a look

**Comment by chraibi** [comment]
> > libc++abi.dylib: terminating with uncaught exception of type boost::numeric::positive_overflow: bad numeric conversion: positive overflow
> > zsh: abort       ini_Methods.xml
> Thanks for testing! I will have a look

@Ozaq: Thanks for looking into this. 

I suspect the culprit is, no surprise here, the geometry. 

In [Analysis.cpp](https://github.com/Ozaq/jpscore/blob/bdef83772ed1e7474c886cd338ae15b20b198e00/jpsreport/src/Analysis.cpp#L257-L260])

the geometry boundaries don't get set properly. 

This leads to polygons having big numbers, which does not work well with `boost::geometry::intersection`

Screenshot 2019-11-01 16 23 08

**Comment by gjaeger** [comment]
> If thats the case I can transfer the missing changes.

This [release](https://github.com/JuPedSim/jpsreport/tree/0.8.5) (0.8.5) would be a version you could adopt.

**Comment by Ozaq** [comment]
> I think, the following points must be taken into account for integration:
> 
>     * [ ]  How do we deal with open issues? Github offers the possibility to transfer them into the repository.
> 
>     * [ ]  Where do we integrate the documentation of jpsreport?
> 
>     * [ ]  How do we manage the release process (different release numbers)?
> 
>     * [ ]  How do we mark issues for jpsreport in jpscore in the future?
> 
>     * [ ]  There are two folders (`demos` and `scripts`) available in both repositories. How do we integrate the jpsreport folders?
> 
>     * [ ]  Who integrates the test cases (Utest) into TeamCity?
After looking into the issues and the current dev speed I propose to deal with this PR in the following way:
Split this into 3 PRs:
1st PR only deals with the jpscore repo to prepare it hosting multiple executables / distinct sub-parts (this is also helpful for the lib required for SUMO)
2nd PR will migrate all of the code from jpsreport unchanged into this repository. An exception here being changes to CMake so that jpsreport can be build from the same project
3rd PR (or - multiple) will deal with refactoring the codebase to eliminate code duplications

As for the points raised above:
* Lets migrate issues after PR2
* Integration of documentation should be part of PR2 -OR- handled in a PR immediately after
* Issues can be labeled based on components, e.g. `libcore`, `jpsreport`, `jpscore`
* As part of the refactoring in PR3+ demo and script folders should also be kept seperate (e.g. as sub folders under jpsreport)
* Integration of the existing tests into TC will probably handled by @zeroset and myself in PR2

Open issue left is releases in my opinion, It al depends on the users and how it is currently distributed. Hard to say for me.

With the plan outlined above we can integrate into one repository and then take it a bit more slowly and careful remove duplications, the issues show above convinced me that this is probably a better approach than to do it on one go and risk introducing errors.

Ill leave this PR here in any case for now

File: jpsreport/src/Analysis.cpp [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+          //search for the subroom that contains that area
+          for (auto&& it_room : _building->GetAllRooms())
+          {
+               for (auto&& it_sub : it_room.second->GetAllSubRooms())
+               {
+                    SubRoom* subroom = it_sub.second.get();
+                    point_2d point(0,0);
+                    boost::geometry::centroid(area->_poly,point);
+                    //check if the area is contained in the obstacle
+                    if(subroom->IsInSubRoom(Point(point.x()/M2CM,point.y()/M2CM)))
```
Comment (chraibi) [inline jpsreport/src/Analysis.cpp]: if(subroom->IsInSubRoomOld(Point(point.x()/M2CM,point.y()/M2CM)))

Comment (chraibi) [reply jpsreport/src/Analysis.cpp]: in `libcore/geometry/SubRoom.cpp` use `bool NormalSubRoom::IsInSubRoom(const Point& ped) const` from `jpsreport` and rename it accordingly.

**Comment by gjaeger** [comment]
@Ozaq and @zeroset: Can we close this PR?

**Pull request closed without merging**